packages/alfa-result/src/err.ts

```diff
@@ -36,30 +36,34 @@ export class Err<E> implements Result<never, E> {
     return true;
   }
 
-  public map(): this {
+  public map(): Err<E> {
     return this;
   }
 
   public mapErr<F>(mapper: Mapper<E, F>): Err<F> {
     return new Err(mapper(this._error));
   }
 
-  public mapOrElse<T, U>(ok: Mapper<T, U>, err: Mapper<E, U>): U {
+  public mapOrElse<U>(ok: unknown, err: Mapper<E, U>): U {
     return err(this._error);
   }
 
-  public flatMap(): this {
+  public apply(): Err<E> {
     return this;
   }
 
-  public reduce<U>(reducer: unknown, accumulator: U): U {
-    return accumulator;
+  public flatMap(): Err<E> {
+    return this;
   }
 
-  public apply(): this {
+  public flatten<T, E>(this: Err<E>): Result<T, E> {
```

Comment:
Should we also have a `flattenErr` doing the symmetric thing? (flatten `Err<Result<T, E>>` and keep `Ok<T>` intact)

Reply:
Errors really shouldn't contain `Ok` values 😬